src/editor/gutter.rs

```diff
@@ -0,0 +1,129 @@
+use crate::{
+    context::PaintCx,
+    cosmic_text::{Attrs, AttrsList, TextLayout},
+    id::Id,
+    peniko::kurbo::Point,
+    view::{AnyWidget, View, ViewData, Widget},
+    Renderer,
+};
+use floem_editor_core::mode::Mode;
+use floem_reactive::RwSignal;
+use kurbo::Rect;
+
+use crate::editor::{color::EditorColor, editor::Editor};
+
+pub struct EditorGutterView {
+    id: Id,
+    data: ViewData,
+    editor: RwSignal<Editor>,
+    width: f64,
+}
+
+pub fn editor_gutter_view(editor: RwSignal<Editor>) -> EditorGutterView {
+    let id = Id::next();
+
+    EditorGutterView {
+        id,
+        data: ViewData::new(id),
+        editor,
+        width: 0.0,
+    }
+}
+
+impl View for EditorGutterView {
+    fn id(&self) -> Id {
```

Comment:
This method doesn't need to be implemented and also you don't need to store both `ViewData` and `Id` as `ViewData` contains the id.